### PR TITLE
gitlab: 12.7.6 -> 12.8.1

### PR DESCRIPTION
**Pull request by talyz**
###### Motivation for this change
Update GitLab to the latest version.

https://about.gitlab.com/releases/2020/02/22/gitlab-12-8-released/
https://about.gitlab.com/releases/2020/02/24/gitlab-12-8-1-released/

This is basically #81057 with fixes and minor additions. I can't add anything to that PR, so I opened this instead.

This still suffers from #79374.

###### Things done
- Tested a few things locally in a VM

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by talyz** [comment]
@GrahamcOfBorg test gitlab

File: pkgs/applications/version-management/gitlab/gitaly/deps.nix [hunk 1]
```diff
@@ -1319,8 +1319,8 @@
     fetch = {
       type = "git";
       url = "https://github.com/ugorji/go";
-      rev = "d75b2dcb6bc8";
-      sha256 = "0di1k35gpq9bp958ywranpbskx2vdwlb38s22vl9rybm3wa5g3ps";
+      rev = "v1.1.4";
```
Comment (petabyteboy) [inline pkgs/applications/version-management/gitlab/gitaly/deps.nix]: I'd like to get a quick confirmation from @flokli that this undoing this change from https://github.com/NixOS/nixpkgs/pull/80062 is fine.

**Comment by petabyteboy** [comment]
Interestingly this fixes #81636, so it would make #81506 unnecessary.

File: pkgs/applications/version-management/gitlab/gitaly/default.nix [hunk 2]
```diff
@@ -16,15 +17,24 @@ let
         };
       };
   };
+  libgit2_0_27 = libgit2.overrideAttrs (oldAttrs: rec {
```
Comment (petabyteboy) [inline pkgs/applications/version-management/gitlab/gitaly/default.nix]: Can you add a note about this fix in the commit description?

**Review by ghost — APPROVED** [review]
Deployed this on a NixOS machine without any problems :+1:

File: pkgs/applications/version-management/gitlab/gitaly/default.nix (hunk 2 above)
Comment (talyz) [reply pkgs/applications/version-management/gitlab/gitaly/default.nix]: Sure!

File: pkgs/applications/version-management/gitlab/gitaly/deps.nix (hunk 1 above)
Comment (talyz) [reply pkgs/applications/version-management/gitlab/gitaly/deps.nix]: Hm, yeah, it seems a little strange..

File: pkgs/applications/version-management/gitlab/gitaly/default.nix (hunk 2 above)
Comment (talyz) [reply pkgs/applications/version-management/gitlab/gitaly/default.nix]: Should be fixed in my latest push.

**Comment by talyz** [comment]
I just added a fix for #79374. It touches the `bundlerEnv` code, but shouldn't affect any other packages than `gitaly` atm.

@GrahamcOfBorg test gitlab

File: pkgs/applications/version-management/gitlab/gitaly/deps.nix (hunk 1 above)
Comment (petabyteboy) [reply pkgs/applications/version-management/gitlab/gitaly/deps.nix]: Seems like it was not done on purpose so it should be fine

Comment (flokli) [reply pkgs/applications/version-management/gitlab/gitaly/deps.nix]: As written there, I don't know why `vgo2nix` it moved to that rev (maybe another package wasn't compatible with it?!), but that rev seems to be older than 1.1.4.

If it now resolves to 1.1.4 again, works for me :-)

**Comment by petabyteboy** [comment]
I deployed the new version and can confirm it fixes https://github.com/NixOS/nixpkgs/issues/79374 too. Thanks!

**Review by flokli — APPROVED** [review]

**Comment by petabyteboy** [comment]
I also ran nix-review to make sure the changes do bundlerEnv do not create problems for other ruby applications.

**Pull request merged**